R/AllClasses2.R

```diff
@@ -532,11 +563,17 @@ output_update <- function(args, dir=TRUE, dir.name=NULL, replace=FALSE, extensio
           dir <- sub("/([^/]*)$", "", args$output[[i]][[j]][k])
           if(grepl(extension[1], name)){
             sam <- strsplit(name, split="\\.")[[1]]
-            sam <- sam[length(sam)]
-            if(sam=="sam"){
-              args$output[[i]][[j]][k] <- suppressWarnings(normalizePath(paste0(dir, "/", names(args$output[i]), extension[2])))
+            useRealNameForSTAR <- ((length(sam)==4) && (sam[2]=="Aligned"))
+            if (useRealNameForSTAR) {
+              realname <- gsub("\\.sam$", "", name)
+              args$output[[i]][[j]][k] <- suppressWarnings(normalizePath(paste0(dir, "/", realname, extension[2])))
             } else {
-              args$output[[i]][[j]][k] <- suppressWarnings(normalizePath(paste0(dir, "/", name)))
+              sam <- sam[length(sam)]
+              if(sam=="sam"){
+                args$output[[i]][[j]][k] <- suppressWarnings(normalizePath(paste0(dir, "/", names(args$output[i]), extension[2])))
+              } else {
+                args$output[[i]][[j]][k] <- suppressWarnings(normalizePath(paste0(dir, "/", name)))
+              }
```

Comment:
Hi @gdmosher, 
is this section to fix the issue about the following?

> When testing star-mapping-pe.cwl (that makes a .sam file) with dir=TRUE and make_bam=TRUE, the files M1A.Aligned.out.bam and bai were not getting renamed into the sample folder. I made this commit specific for STAR to prevent any other workflows from being affected. The solution can be made more generic if you like it.